scripts/zones/Tavnazian_Safehold/npcs/Yurim.lua

```diff
@@ -1,301 +1,132 @@
-    if (reward == true) then
-        player:startEvent(529, GORGET, SILVER_EARRING, SILVER_OBI);
     end
-end;
+end
 
 function onTrigger(player,npc)
+    -- IN THE NAME OF SCIENCE
+    if player:hasCompletedMission(COP, THE_WARRIOR_S_PATH) then
+        local nameOfScience  = player:getQuestStatus(OTHER_AREAS_LOG,IN_THE_NAME_OF_SCIENCE)
+        local itemInProgress = player:getVar("NAME_OF_SCIENCE_target")
 
-    local NameOfScience = player:getQuestStatus(OTHER_AREAS_LOG,IN_THE_NAME_OF_SCIENCE);
-    local itemtarget = player:getVar("NAME_OF_SCIENCE_target");
-    local rnd= math.random();
-
-    if (player:getCurrentMission(COP) > THE_WARRIOR_S_PATH) then   --have SEA acces
-        if (NameOfScience == QUEST_AVAILABLE) then
-            player:startEvent(524,13205,13327,13080);
-            player:addQuest(OTHER_AREAS_LOG,IN_THE_NAME_OF_SCIENCE);
-        elseif (NameOfScience == QUEST_ACCEPTED or NameOfScience ==QUEST_COMPLETED  and itemtarget == 0) then
-            player:startEvent(525,13205,13327,13080);
-        elseif (NameOfScience == QUEST_ACCEPTED or NameOfScience ==QUEST_COMPLETED and itemtarget > 0) then
-            if (rnd < 0.3) then
-                player:startEvent(532,getNOS(itemtarget,1),getNOS(itemtarget,2));
+        if nameOfScience == QUEST_AVAILABLE then
+            player:startEvent(524, obi, earring, gorget)
+        elseif (nameOfScience == QUEST_ACCEPTED or nameOfScience == QUEST_COMPLETED) and itemInProgress == 0 then
+            player:startEvent(525, obi, earring, gorget)
+        elseif nameOfScience == QUEST_ACCEPTED or nameOfScience == QUEST_COMPLETED then
+            if math.random(100) <= 30 then
+                player:startEvent(532, unpack(nosTrades[itemInProgress].base))
             else
-                player:startEvent(528,getNOS(itemtarget,1),getNOS(itemtarget,2));
+                player:startEvent(528, unpack(nosTrades[itemInProgress].base))
             end
         end
+
+    -- STANDARD DIALOG
     else
-        if (rnd < 0.5) then
-            player:startEvent(527); --no quest
+        if math.random(100) <= 50 then
+            player:startEvent(527)
         else
-            player:startEvent(519); --no quest
+            player:startEvent(519)
         end
     end
-end;
+end
 
 function onEventUpdate(player,csid,option)
-end;
+end
 
 function onEventFinish(player,csid,option)
-    if (csid == 529) then
-        local itemtarget = player:getVar("NAME_OF_SCIENCE_target");
-        if (player:getFreeSlotsCount() == 0) then
-            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED,itemtarget);
+    if csid == 524 then
+        player:addQuest(OTHER_AREAS_LOG, IN_THE_NAME_OF_SCIENCE)
+    elseif csid == 531 then
+        player:confirmTrade()
+    elseif csid == 526 then
+        player:confirmTrade()
+    elseif csid == 529 then
+        local itemInProgress = player:getVar("NAME_OF_SCIENCE_target")
+        local itemComplete   = player:getVar("NAME_OF_SCIENCE_complete")
```

Comment:
we should be able to accomplish this using getLocalVar/setLocalVar

Reply:
Changed the anti-cheat one, but the in-progress var must remain as setVar since player could log out after trading the base items.